pkg/cache/cache_test.go

```diff
@@ -721,34 +796,71 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					Expect(namespacedCache.Get(context.Background(), key2, node)).To(Succeed())
 				})
 
-				It("should deep copy the object unless told otherwise", func() {
-					By("retrieving a specific pod from the cache")
-					out := &metav1.PartialObjectMetadata{}
-					out.SetGroupVersionKind(schema.GroupVersionKind{
-						Group:   "",
-						Version: "v1",
-						Kind:    "Pod",
+				if opts.UnsafeDisableCacheDeepCopy == nil {
+					It("should deep copy the object unless told otherwise", func() {
+						By("retrieving a specific pod from the cache")
+						out := &metav1.PartialObjectMetadata{}
+						out.SetGroupVersionKind(schema.GroupVersionKind{
+							Group:   "",
+							Version: "v1",
+							Kind:    "Pod",
+						})
+						uKnownPod2 := &metav1.PartialObjectMetadata{}
+						knownPod2.(*corev1.Pod).ObjectMeta.DeepCopyInto(&uKnownPod2.ObjectMeta)
+						uKnownPod2.SetGroupVersionKind(schema.GroupVersionKind{
+							Group:   "",
+							Version: "v1",
+							Kind:    "Pod",
+						})
+
+						podKey := client.ObjectKey{Name: "test-pod-2", Namespace: testNamespaceTwo}
+						Expect(informerCache.Get(context.Background(), podKey, out)).To(Succeed())
+
+						By("verifying the retrieved pod is equal to a known pod")
+						Expect(out).To(Equal(uKnownPod2))
+
+						By("altering a field in the retrieved pod")
+						out.Labels["foo"] = "bar"
+
+						By("verifying the pods are no longer equal")
+						Expect(out).NotTo(Equal(knownPod2))
 					})
-					uKnownPod2 := &metav1.PartialObjectMetadata{}
-					knownPod2.(*corev1.Pod).ObjectMeta.DeepCopyInto(&uKnownPod2.ObjectMeta)
-					uKnownPod2.SetGroupVersionKind(schema.GroupVersionKind{
-						Group:   "",
-						Version: "v1",
-						Kind:    "Pod",
+				} else {
+					It("should not deep copy the object if UnsafeDisableCacheDeepCopy is enabled", func() {
+						By("getting a specific pod from the cache twice")
+						podKey := client.ObjectKey{Name: "test-pod-2", Namespace: testNamespaceTwo}
+						out1 := &metav1.PartialObjectMetadata{}
+						out1.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"})
+						Expect(informerCache.Get(context.Background(), podKey, out1)).To(Succeed())
+						out2 := &metav1.PartialObjectMetadata{}
+						out2.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"})
+						Expect(informerCache.Get(context.Background(), podKey, out2)).To(Succeed())
+
+						By("verifying the pods have the same pointer addresses")
+						By("verifying the pointer fields in pod have the same addresses")
+						Expect(out1).To(Equal(out2))
+						Expect(reflect.ValueOf(out1.Labels).Pointer()).To(BeIdenticalTo(reflect.ValueOf(out2.Labels).Pointer()))
```

Comment:
non-blocking nit: pretty sure there's no need to actually fetch the pointer for `BeIdenticalTo`. This could be simplified to:

```suggestion
						Expect(out1.Labels).To(BeIdenticalTo(out2.Labels))
```

Reply:
I tried `Expect(out1.Labels).To(BeIdenticalTo(out2.Labels))` before, but it failed. I'm not sure why this happened, the `out1.Labels` and `out2.Labels` do have the same pointer addresses.

Reply:
Hmm, alright 🤷‍♂️. Thanks for checking that out!